app/src/main/java/de/danoeh/antennapod/adapter/CoverLoader.java

```diff
@@ -132,11 +139,27 @@ protected void onResourceCleared(@Nullable Drawable placeholder) {
             ivCover.setImageDrawable(placeholder);
         }
 
-        static void setPlaceholderVisibility(TextView placeholder, boolean textAndImageCombined) {
+        static void setPlaceholderVisibility(TextView placeholder, boolean textAndImageCombined, Bitmap bitmap) {
+            boolean showTitle = UserPreferences.shouldShowSubscriptionTitle();
             if (placeholder != null) {
-                if (textAndImageCombined) {
+                if (textAndImageCombined || showTitle) {
                     int bgColor = placeholder.getContext().getResources().getColor(R.color.feed_text_bg);
-                    placeholder.setBackgroundColor(bgColor);
+                    if (bitmap == null || !showTitle) {
+                        placeholder.setBackgroundColor(bgColor);
+                        return;
+                    }
+                    Palette.from(bitmap).generate(p -> {
+                        if (p == null) {
+                            return;
+                        }
+                        int dominantColor = p.getDominantColor(bgColor);
+                        int textColor = placeholder.getContext().getResources().getColor(R.color.white);
+                        if (ColorUtils.calculateLuminance(dominantColor) > 0.6) {
```

Comment:
I would do `0.5` here. The text on one of my test subscriptions is pretty hard to read.

<img width="100" src="https://user-images.githubusercontent.com/5811634/142283792-323a7ba9-240f-4f5f-8a3f-40eb1c4aaa5b.png" />